### ISSUE TITLE
Lua file's load() called automatically and before Love.Scene.Load()

### ISSUE THREAD

**Issue by matej-zajacik** [endlesstravel/Love2dCS#57]
When you enable the Lua module and specify the main file to load up in `BootConfig`, the framework will initialize the Lua module and call the Lua file's `love.load()` function (if it's provided) **before** the C#'s side of things -> `Love.Scene.Load()`.

This is unexpected.

The other two backbone callbacks supported for Lua, `Love.Lua.Update()` and `Love.Lua.Draw()`, are supposed to be manually invoked from C#. And that is a good thing, in my opinion. You are writing a C# app and you should be in control of what gets called when. I see Lua as a service to the app, so the app should decide on this.

I think the same rule should apply to the `Load()` function. In essence, you would do this:

```C#
public override void Load()
{
  Love.Lua.Load();
}
```

...which would invoke the `love.load()` function in the main Lua file.

The advantage is that you would be able to call it in an order specified by you. Maybe you need to initialize some C# stuff first, and only then initialize the Lua side of things.

**Notes:**

I'm not sure what to do with the file scope of the Lua file, which runs even before the `load()` callback, obviously.

**Comment by matej-zajacik** [comment]
Excellent!

Regarding the docs about how to use Lua, I'll wait for when you release this new functionality, and then I'll tweak the page to be accurate. Right now, the page is mostly done (locally), but would soon become obsolete.

**Comment by matej-zajacik** [comment]
Excellent!

Regarding the docs about how to use Lua, I'll wait for when you release this new functionality, and then I'll tweak the page to be accurate.

**Comment by endlesstravel** [comment]
i improve lua in version [11.0.27](https://www.nuget.org/packages/Love2dCS/11.0.27)

it now look like:
```C#
public class IS57 :Scene
{
    public override void Load()
    {
        Lua.Load("Content/my_main.lua");

        Lua.RegisterFunction(typeof(Math), "Cos", "cos");
        Lua.RegisterFunction(typeof(IS57), "StaticWriteLine", "staticWriteLine");
        Lua.RegisterFunction(this, "Add", "add");
        Lua.RegisterFunction(this, "PrintLine", "printLine");
        Lua.RegisterFunction(this, "PrintArrayLine", "printArrayLine");
        Lua.RegisterFunction(this, "PrintArrayLineFloat", "printArrayLineFloat");
        Lua.RegisterFunction(this, "ByteToStr");

        Lua.DoString(" love.sharp.staticWriteLine('staticWriteLine') ");

        Console.WriteLine("Math.Cos(3.14f / 2f)  : " + Math.Cos(3.14f / 2f));
        Lua.DoString(" love.sharp.staticWriteLine('love.sharp.cos(3.14 / 2) : ' ..  love.sharp.cos(3.14 / 2)) ");
        Lua.DoString(" love.sharp.printLine('----------------------------') ");
        Lua.DoString(" love.sharp.printLine('byte to str  : ' ..  love.sharp.ByteToStr(1)) ");
        Lua.DoString(" love.sharp.printLine(love.sharp.add(123, 456)) ");
        Lua.DoString(" love.sharp.printArrayLine({ 'aaa', 'bbb' }) ");
        Lua.DoString(" love.sharp.printArrayLineFloat({ 1, 2 }) ");
    }

    public override void Update(float dt)
    {
        Lua.Update(dt);
    }
    public override void Draw()
    {
        Lua.Draw();
    }

    public static void StaticWriteLine(string info) => Console.WriteLine(info);
    public string ByteToStr(byte b) => b.ToString();
    public void PrintArrayLineFloat(float[] info) => Console.WriteLine(string.Join(",  ", info));
    public void PrintArrayLine(string[] info) => Console.WriteLine(string.Join(",  ", info));
    public void PrintLine(string info) => Console.WriteLine(info);
    public int Add(int a, int b) => a + b;
}

```
and the result is :

![image](https://user-images.githubusercontent.com/10757838/57582680-4fbdde00-74fa-11e9-89df-0df5199be3fd.png)

**Comment by matej-zajacik** [comment]
In 11.0.27, all of the described above works, but there is another issue now. The `love.load` in the Lua file is not invoked by calling the `Love.Lua.Load` from C#. The file itself is loaded and evaluated (the file scope / chunk), but not the `load` function.

I think one would expect the C# `Load` to invoke the Lua `load`. What do you think?

**Comment by endlesstravel** [comment]
oh, you are right, my original plan is  automatic call `love.load` when invode `Love.Lua.Load`. 
I forget add invoke `love.load` ! 😅 
i will fix this  !

**Comment by endlesstravel** [comment]
bug that `love.load` not invoked has fixed in [11.0.29](https://www.nuget.org/packages/Love2dCS/11.0.29)
